src/systemctl/systemctl-edit.c

```diff
@@ -432,6 +433,22 @@ static int find_paths_to_edit(
         assert(names);
         assert(ret_edit_files);
 
+        if (!arg_drop_in || streq(arg_drop_in, ""))
+                dropin = strdup("override.conf");
+        else if (!endswith(arg_drop_in, ".conf"))
+                dropin = strjoin(arg_drop_in, ".conf");
+        else
+                dropin = strdup(arg_drop_in);
```

Comment:
You can slightly optimize the memory allocation here:
```c
_cleanup_free_ char *dropin_alloc = NULL;
const char *dropin;

...

if (isempty(arg_drop_in))
        dropin = "override.conf";
else if (endswith(arg_drop_in, ".conf"))
        dropin = arg_drop_in;
else {
        dropin_alloc = strjoin(arg_drop_in, ".conf");
        if (!dropin_alloc)
                return log_oom();
        dropin = dropin_alloc;
}
```
BTW, not a big matter, but better to name `drop_in` instead of `dropin`, for consistency with `arg_drop_in` and so on.